nncf/common/graph/graph.py

```diff
@@ -143,7 +143,7 @@ def __init__(
         self.to_node = to_node
         self.input_port_id = input_port_id
         self.output_port_id = output_port_id
-        self.tensor_shape = tensor_shape
+        self.tensor_shape: Tuple[int] = tuple(tensor_shape)
```

Comment:
What is the reason for these changes?

Reply:
Type hinting, and through experience I have found that using tuples to represent shapes works better since there is no way to accidentally store a shape by reference where you expected it to be stored by copy.

Reply:
I think it makes sense to use a tuple here because we probably don't want to change the shape at the edge. However, I think it's better to change the type in the signature of the `NNCFGraphEdge.__init__()` method.

Reply:
I can do this if you insist, but I think that the most points in code right now try to pass a list to the __init__ method, so changing the __init__ signature would require changes in all of those places too if we want strict type conformance in this spot.

Reply:
I'm okay with it. I don't think it is necessary to do this here.